### PR TITLE
Improved calib type 6 w/ FAP.

### PR DESCRIPTION
**Pull request by MSECode**
Tested and working on both desk setup and ergocab robot.

In this PR the following tasks have been done:
- Improve current working calibration type 6 by allowing to use FAP sensors instead of MICE;
- Tested the calibration procedure on the left arm bench setup.
- Tested the calibration procedure on the left arm of ergocub robot (test on right arm is going to be done)
- Updated firmware code by customizing calibration type 6 enabling POS service use

### CODE REVIEW

**Comment by marcoaccame** [comment]
Hi @MSECode, welcome to your first PR in robotology/icub-firmware. I will have a look at it later today.

File: emBODY/eBcode/arch-arm/embobj/plus/mc/Calibrators.c [hunk 1]
```diff
@@ -112,7 +112,7 @@ static eOresult_t JointSet_do_wait_calibration_6_singleJoint(JointSet *o, int in
                 
                 jCalib6Data_ptr->is_active = FALSE;
                 char info[80];
-                sprintf(info,"calib 6:outLim: cp%d mx%.1f mn%.1f",curr_pos, j_ptr->pos_max, j_ptr->pos_min);
+                sprintf(info,"calib 6:outLim: cp:%d mx:%.1f mn:%.1f",curr_pos, j_ptr->pos_max, j_ptr->pos_min);
```
Comment (marcoaccame) [inline emBODY/eBcode/arch-arm/embobj/plus/mc/Calibrators.c:115]: just an improvement: `snprintf(info, sizeof(info), `

File: emBODY/eBcode/arch-arm/embobj/plus/mc/Calibrators.c [hunk 2]
```diff
@@ -210,11 +205,11 @@ static eOresult_t JointSet_do_wait_calibration_6_singleJoint(JointSet *o, int in
...
                 m_ptr->pos_min = j_ptr->running_calibration.data.type6.rotorposmin;
                 m_ptr->pos_max = j_ptr->running_calibration.data.type6.rotorposmax;
-//                /////Debug code 
-//                char info[80];
-//                sprintf(info,"calib 6: restore rlim of m %d", m_ptr->ID);
-//                JointSet_send_debug_message(info, j_ptr->ID);
-//                ////ended
+                /////Debug code 
+                char info[80];
+                sprintf(info,"m_min=%d m_max=%d", m_ptr->pos_min, m_ptr->pos_max );
```
Comment (marcoaccame) [inline emBODY/eBcode/arch-arm/embobj/plus/mc/Calibrators.c:210]: just an improvement: `snprintf(info, sizeof(info), `

File: emBODY/eBcode/arch-arm/embobj/plus/mc/Calibrators.c [hunk 3]
```diff
@@ -223,6 +218,9 @@ static eOresult_t JointSet_do_wait_calibration_6_singleJoint(JointSet *o, int in
         case calibtype6_st_finished:
         {
             *calibrationCompleted = TRUE;
+            char info[80];
+            sprintf(info,"calib Completed" );
```
Comment (marcoaccame) [inline emBODY/eBcode/arch-arm/embobj/plus/mc/Calibrators.c:222]: just an improvement: `snprintf(info, sizeof(info), `

File: emBODY/eBcode/arch-arm/embobj/plus/mc/JointSet.c [hunk 4]
```diff
@@ -1562,6 +1572,11 @@ void JointSet_calibrate(JointSet* o, uint8_t e, eOmc_calibrator_t *calibrator)
                 return;
             }
             
+            ////debug code
+            char info[50];
+            snprintf(info, 50, "vmax=%d,vim=%d",calibrator->params.type6.vmax, calibrator->params.type6.vmin);
```
Comment (marcoaccame) [inline emBODY/eBcode/arch-arm/embobj/plus/mc/JointSet.c:1577]: just an improvement: `snprintf(info, sizeof(info), `

**Review by marcoaccame — CHANGES_REQUESTED** [review]
The PR is fine. I assume it was tested on the robot. 
I have just noted the `snprintf` but they are very minor things. 

Thanks for your contribution.

**Comment by marcoaccame** [comment]
Companion PR is: 
- https://github.com/robotology/icub-firmware-build/pull/71

File: emBODY/eBcode/arch-arm/embobj/plus/mc/Calibrators.c (hunk 1 above)
Comment (MSECode) [reply emBODY/eBcode/arch-arm/embobj/plus/mc/Calibrators.c:115]: Oks, I'll fix that and I'll submit the change

**Comment by MSECode** [comment]
Submitted changes in [commit ](https://github.com/valegagge/icub-firmware/commit/e3df97a6598d1859cc780e358349f3542a761d46)

**Review by marcoaccame — APPROVED** [review]

**Pull request merged**